include/seqan3/io/stream/parse_condition.hpp

```diff
@@ -0,0 +1,1435 @@
+};
+//!\endcond
+
+/*!\name Requirements for seqan3::detail::parse_condition_concept
+ * \relates seqan3::detail::parse_condition_concept
+ * \brief You can expect the variable and the predicate function on all types that satisfy seqan3::ostream_concept.
+ * \{
+ */
+/*!\fn      bool operator()(char_type c);
+ * \brief   Predicate function to test if `c` satisfies the given condition.
+ * \ingroup stream
+ * \param   c The character to be tested.
+ * \returns `true` on success, `false` otherwise.
+ *
+ * \attention This is a concept requirement, not an actual function (however types satisfying this concept
+ * will provide an implementation).
+ */
+
+/*!\var static constexpr auto msg
+ * \memberof seqan3::detail::parse_condition_concept
+ * \brief Defines the condition msg. The type is deduced from the constant expression in the definition of the variable.
+ */
+//!\}
+
+// ----------------------------------------------------------------------------
+// make_printable
+// ----------------------------------------------------------------------------
+
+/*!\brief Returns a printable value for the given character `c`.
+ * \param[in] c The character to be represented as printable string.
+ * \return    a std::string containing a printable version of the given character `c`.
+ *
+ * \details
+ *
+ * Some characters, e.g. control commands cannot be printed. This function converts them to a std::string
+ * containing the visual representation of this character. For all control commands the value `'CTRL'` is returned.
+ *
+ * ### Exception
+ *
+ * Strong exception guarantee is given.
+ *
+ * ### Complexity
+ *
+ * Constant.
+ *
+ * ### Concurrency
+ *
+ * Thread-safe.
+ */
+template <typename char_type>
+inline std::string
+make_printable(char_type const c)
+{
+    if (c == '\0')
+        return "'\0'";
+    else if (c == '\t')
+        return "'\t'";
+    else if (c == '\n')
+        return "'\n'";
+    else if (c == '\v')
+        return "'\v'";
+    else if (c == '\f')
+        return "'\f'";
+    else if (c == '\r')
+        return "'\r'";
+    else if ((c >= static_cast<char>(1) && c <= static_cast<char>(8)) ||
+                  (c >= static_cast<char>(14) && c <= static_cast<char>(31)))
+        return "'CTRL'";
+    else if (c == static_cast<char>(127))
+        return "'DEL'";
+    else
+        return {'\'', c, '\''};
+}
+
+// ----------------------------------------------------------------------------
+// parse_condition
+// ----------------------------------------------------------------------------
+
+//!\cond
+template <parse_condition_concept... condition_ts>
+    requires sizeof...(condition_ts) >= 2
+struct parse_condition_combiner;
+
+template <parse_condition_concept condition_t>
+struct parse_condition_negator;
+//!\endcond
+
+/*!\brief A [CRTP](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern) class for parse conditions to add
+ *        logical conjunction, disjunction and negation operator.
+ * \ingroup stream
```

Comment:
I also I find it intuitive to always call the template argument of a CRTP base `derived_t`, because it's very obvious when reading that this is the derived type. When read `condition_t` further down, I first have to scroll up to see that this isn't another template parameter... but it's not soo important maybe...

Reply:
Inheritance from the parse_condition is a requirement for the helper concept. 
It probably could have been a standard inheritance class, since the condition types are stateless and data slicing would not be an issue. But I think it is better to keep the design like in other places, in order to avoid mixtures of designs. 
Though, the point with making the CRTP base class abstract, is actually a very good.